src/ids.rs

```diff
@@ -54,11 +79,62 @@ impl PyIds {
         vm: &PyVM,
         references: &HashMap<String, HintReference>,
         ap_tracking: &ApTracking,
+        struct_types: Rc<HashMap<String, HashMap<String, Member>>>,
     ) -> PyIds {
         PyIds {
             vm: vm.get_vm(),
             references: references.clone(),
             ap_tracking: ap_tracking.clone(),
+            struct_types,
+        }
+    }
+}
+
+#[pyclass(unsendable)]
+struct PyTypedId {
+    vm: Rc<RefCell<VirtualMachine>>,
+    hint_value: Relocatable,
+    cairo_type: String,
+    struct_types: Rc<HashMap<String, HashMap<String, Member>>>,
+}
+
+#[pymethods]
+impl PyTypedId {
+    #[getter]
+    fn __getattr__(&self, py: Python, name: &str) -> PyResult<PyObject> {
+        let struct_type = self.struct_types.get(&self.cairo_type).unwrap();
+        if name == "address_" {
+            return Ok(PyMaybeRelocatable::from(self.hint_value.clone()).to_object(py));
+        }
+        match struct_type.get(name) {
+            Some(member) => {
+                let vm = self.vm.borrow();
+                Ok(match member.cairo_type.as_deref() {
+                    Some("felt") | Some("felt*") => PyMaybeRelocatable::from(
+                        vm.get_maybe(
+                            &self
+                                .hint_value
+                                .add_int_mod(member.offset.as_ref().unwrap(), vm.get_prime())
+                                .map_err(to_py_error)?,
+                        )
+                        .map_err(to_py_error)?
+                        .unwrap(),
+                    )
+                    .to_object(py),
+                    Some(cairo_type) => PyTypedId {
+                        vm: self.vm.clone(),
+                        hint_value: self.hint_value.clone(),
+                        cairo_type: cairo_type.to_string(),
+                        struct_types: self.struct_types.clone(),
+                    }
+                    .into_py(py),
+                    None => todo!(),
```

Comment:
This should either be `unreachable!` or an error.

Reply:
And if it happens to be an error, then maybe we shouldn't use an option I think.